From 82a096ede0d36b69c60285e172cd11cd4a69dd85 Mon Sep 17 00:00:00 2001
From: Marc Zyngier <marc.zyngier@arm.com>
Date: Wed, 16 Dec 2015 11:03:24 +0000
Subject: [PATCH 003/147] irqchip/gic-v2m: Fix of_node refcount on error

Git-commit: 86d14c72b7837589a4381b3bc2e117e7d842a92a
Patch-mainline: v4.5-rc1
References: fate#320512, fate#322061

On the error path, the v2m drivers drops the refcount on the parent
node instead of doing it on the node that generated the error.
Humph...

Reported-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/irqchip/irq-gic-v2m.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-gic-v2m.c b/drivers/irqchip/irq-gic-v2m.c
index 87f8d10..ee1e553 100644
--- a/drivers/irqchip/irq-gic-v2m.c
+++ b/drivers/irqchip/irq-gic-v2m.c
@@ -389,7 +389,7 @@ int __init gicv2m_of_init(struct device_node *node, struct irq_domain *parent)
 
 		ret = gicv2m_init_one(child, parent);
 		if (ret) {
-			of_node_put(node);
+			of_node_put(child);
 			break;
 		}
 	}
-- 
1.8.5.6

